libusb/descriptor.c

```diff
@@ -32,16 +32,16 @@
 
 static inline uint16_t ReadLittleEndian16(const uint8_t p[2])
 {
-	return (uint16_t)p[1] << 8 |
-	       (uint16_t)p[0];
+	return (uint16_t)((uint16_t)p[1] << 8 |
+					  (uint16_t)p[0]);
 }
 
 static inline uint32_t ReadLittleEndian32(const uint8_t p[4])
 {
-	return (uint32_t)p[3] << 24 |
-	       (uint32_t)p[2] << 16 |
-	       (uint32_t)p[1] << 8 |
-	       (uint32_t)p[0];
+	return (uint32_t)((uint32_t)p[3] << 24 |
+					  (uint32_t)p[2] << 16 |
+					  (uint32_t)p[1] << 8 |
+					  (uint32_t)p[0]);
```

Comment:
yeah, I usually hate mixed indentation, but this is exactly the place for it:
```suggestion
	return (uint32_t)((uint32_t)p[3] << 24 |
	                  (uint32_t)p[2] << 16 |
	                  (uint32_t)p[1] << 8 |
	                  (uint32_t)p[0]);
```